### PR TITLE
[Keymap] moults31 userspace and moults31 keymap on multiple keyboards

### PR DESCRIPTION
**Pull request by moults31**
## Added keymap.c and readme.

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/dztech/dz60rgb/keymaps/moults31/keymap.c [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/dztech/dz60rgb/keymaps/moults31/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 2]
```diff
@@ -55,6 +55,7 @@ doxygen/
 .vscode/c_cpp_properties.json
 .vscode/launch.json
 .vscode/tasks.json
+.vscode/settings.json
```
Comment (drashna) [inline .gitignore:58]: Please revert this change. 
```suggestion
```

Comment (moults31) [reply .gitignore]: Thanks, fixed in 61a855f785e52948c152849148cd094652e8fb3e

File: keyboards/dztech/dz60rgb/keymaps/moults31/keymap.c (hunk 1 above)
Comment (moults31) [reply keyboards/dztech/dz60rgb/keymaps/moults31/keymap.c:18]: Thanks, fixed in 61a855f785e52948c152849148cd094652e8fb3e

**Comment by moults31** [comment]
@drashna  - Thanks for the feedback, I have resolved the two items in the latest commit.

Note that I also changed the title to reflect the scope creep that occurred while this PR was stale.

Once this is merged, my future PRs will be much more focused :)

File: users/moults31/vscode.c [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+ */
+#include "vscode.h"
+
+bool process_record_vsc(uint16_t keycode, keyrecord_t *record) {
+    bool rv = true;
+
+    switch (keycode) {
+        case M_VSC_TERMFOCUS:
+            if (record->event.pressed) {
+                SEND_STRING(SS_LCTRL("`"));
```
Comment (drashna) [inline users/moults31/vscode.c:24]: Just a heads up, most of these could be done with simple modded keycodes. 

Eg `C(KC_GRV)`, or `C(S(KC_2))`.   And the `tap_code16` function will support these, too.

Comment (moults31) [reply users/moults31/vscode.c:24]: Thanks for the heads-up, I'll definitely be going back through and cleaning these up sooner than later

File: keyboards/knobgoblin/knobgoblin.c [hunk 4]
```diff
@@ -80,24 +80,23 @@ __attribute__((weak)) void oled_task_user(void) {
 
 	switch (get_highest_layer(layer_state)) {
 		case 0:
-			oled_write_P(PSTR(" ONE\n"), false);
+			oled_write_P(PSTR(" NUM\n"), false);
```
Comment (drashna) [inline keyboards/knobgoblin/knobgoblin.c:83]: please revert these changes. 

If needed, copy the whole function and past it in your keymap

**Review by drashna — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Comment by drashna** [comment]
Thanks!
